### ISSUE TITLE
Support for Edge and IE11

### ISSUE THREAD

**Issue by holtjonathan** [okta/okta-oidc-js#604]
## I'm submitting this issue for the package(s):

- [ ] jwt-verifier
- [ ] okta-angular
- [ ] oidc-middleware
- [X] okta-react
- [ ] okta-react-native
- [ ] okta-vue

## I'm submitting a:

- [X] Bug report  
- [ ] Feature request
- [ ] Other (Describe below)

## Current behavior
For Edge, the demo simply does nothing when the Login button is pressed.  I have confirmed that the async login() method is being called but when it further tries to call "this.props.auth.login("/")" nothing happens.  No error in the console or anything to track.  It is like the auth.login() method is simply doing nothing.

For IE11, the demo apps just do not work at all.  In fact, IE11 throws the error "AuthSdkError: This browser doesn't support PKCE".  So does Okta just not support IE11 while using React?

## Expected behavior
That I can log in with both Edge and IE11.

## Minimal reproduction of the problem with instructions
Simply clone the react okta repo and run it in IE11 and Edge.  WIth IE11 you will immediately receive errors and with Edge you will see that the login button does nothing.

## Environment

- Package Version:
- Browser: IE11/Edge
- OS:
- Node version (`node -v`):
- Other:

**Comment by holtjonathan** [comment]
Any ideas about these issues?  This is preventing my boss from agreeing that Okta is what way to go.  He doesn't want to pick something that clearly doesn't support Edge/IE11.

**Comment by swiftone** [comment]
@holtjonathan - Thanks for the question.  Here are some details:

For IE11, I suggest you look at the comments here: https://github.com/okta/okta-oidc-js/issues/591

TL;DR: IE11 doesn't support secure webcrypto, which is required for PKCE.  You can polyfill it (knowing that the polyfill WON'T be as secure as legit webcrypto) or not use PKCE.  Those comments give the means to check for PKCE support.  

Edge should work fine, and we've not gotten any reports of it not working.  Can you provide details for a minimal reproduction case?

**Comment by holtjonathan** [comment]
I'll take a look at the link you provided for IE11.  Thanks for that.

For Edge, I cloned the repo and try to run it.  In Chrome, all works as intended.  In IE11 nothing works (due to the previously mentioned PKCE problem).  But in Edge, it just does nothing.  The props.auth.login() method doesn't route me over to the Okta login page.  The button appears to do nothing.

**Comment by holtjonathan** [comment]
@swiftone  If I set `pkce: false` then IE11 works the same as Edge.  The page loads but the Login button does nothing.  The user isn't being routed to the Okta login page like they are while using Chrome.

Forgive my lack of understand (and probably a stupid question :-)), but what is the purpose of PKCE?  Isn't Okta already fairly secured?  Does this just add another layer on top of it?

**Comment by holtjonathan** [comment]
@swiftone  Wow, thank you for that answer.  Really helped explain alot.  I'll check out the link that you provided.

Any chance that you were able to recreate the Edge problem though?  I am seeing nothing happen in Edge/IE11 (with PKCE disabled) so I do not know the next step that I should take.

**Comment by holtjonathan** [comment]
I was able to get Edge to work like Chrome after disabling PKCE.  While IE11 no longer immedately throws the error about not supporting PKCE, it now behaves like Edge used to (clicking 'Login' doesn't route the user to the okta login page).  

It seems like in order to support Edge, I have to disable PKCE but either way IE11 does not work.

**Comment by swiftone** [comment]
I just discovered this issue: https://github.com/okta/okta-auth-js/issues/276 which may shed some light on your experience with Edge.  We have some work in progress (see https://github.com/okta/okta-auth-js/pull/284 ) to hopefully correct that.

Thanks for sharing your experiences!

**Comment by holtjonathan** [comment]
That is great!  Sounds exactly like my problem.  Thank you for linking those.  I'll go ahead and close this since those are in the process of getting fixed.
